File: Q1/QAP-4-Files-AO/Javascript-QAP4/yacht.js
// Desc: Prepare customer receipt for St. John’s Marina & Yacht Club.
// Author: Abiodun Magret Oyedele & SD 14
// Dates: March 22 2025

// Define program constants.
const SITE_COST_EVEN = 80.00;
const SITE_COST_ODD = 120.00;
const ALT_MEMBER_COST = 5.00;
const CLEANING_COST = 50.00;
const SURVEILLANCE_COST = 35.00;
const HST_RATE = 0.15; // HST rate (15%)
const MONTHLY_DUES_STANDARD = 75.00;
const MONTHLY_DUES_EXECUTIVE = 150.00;
const PROCESSING_FEE = 59.99;
const CANCELLATION_RATE = 0.6;

// Define format options for printing.
const cur2Format = new Intl.NumberFormat("en-CA", {
  style: "currency",
  currency: "CAD",
  minimumFractionDigits: 2,
  maximumFractionDigits: 2,
});

const per0Format = new Intl.NumberFormat("en-CA", {
  style: "percent",
  minimumFractionDigits: 0,
  maximumFractionDigits: 0,
});

// Gather user input.
let currentDate = prompt("Enter the current date (YYYY-MM-DD): ");
let siteNumber = parseInt(prompt("Enter the site number (1-100): "));
let memberName = prompt("Enter the member's name: ");
let streetAddress = prompt("Enter the street address: ");
let city = prompt("Enter the city: ");
let province = prompt("Enter the province: ");
let postalCode = prompt("Enter the postal code: ");
let phoneNumber = prompt("Enter the phone number: ");
let cellNumber = prompt("Enter the cell number: ");
let membershipType = prompt("Enter membership type (S for Standard, E for Executive): ").toUpperCase();
let alternateMembers = parseInt(prompt("Enter the number of alternate members: "));
let weeklyCleaning = prompt("Weekly site cleaning? (Y for Yes, N for No): ").toUpperCase();
let videoSurveillance = prompt("Video surveillance? (Y for Yes, N for No): ").toUpperCase();

// Perform program calculations.
let siteCharges = 0;
if (siteNumber % 2 === 0) {
    siteCharges = SITE_COST_EVEN + (alternateMembers * ALT_MEMBER_COST);
} else {
    siteCharges = SITE_COST_ODD + (alternateMembers * ALT_MEMBER_COST);
}

let extraCharges = 0;
if (weeklyCleaning === "Y") {
    extraCharges += CLEANING_COST;
} else if (videoSurveillance === "Y") {
    extraCharges += SURVEILLANCE_COST;
}

let subTotal = siteCharges + extraCharges;
let tax = subTotal * HST_RATE;
let totalMonthlyCharge = subTotal + tax

// monthly dues
let monthlyDues = 0
if (membershipType === "S") {
    monthlyDues += MONTHLY_DUES_STANDARD;
} else if (membershipType === "E") {
    monthlyDues += MONTHLY_DUES_EXECUTIVE;
}

let totalMonthlyFees = totalMonthlyCharge + monthlyDues;
let totalYearlyFees = totalMonthlyFees * 12

let monthlyPayment = (totalYearlyFees + PROCESSING_FEE) / 12;

let cancellationFee = (siteCharges * 12) * CANCELLATION_RATE;


// Display program results to the screen in a table.
// I found it challenging to align the text to the right without using a divider between the labels and values.
// Additionally, using &nbsp; for spacing is problematic when the amount of data varies, such as when 'No' has
// 2 characters and 'Yes' has 3, making it difficult to maintain consistent alignment.
document.writeln("<br />");
document.writeln("<table class='receipttable'>");

document.writeln("<tr>");
document.writeln("<td>St. John’s Marina & Yacht Club<br />Yearly Member Receipt</td>");
document.writeln("</tr>");

document.writeln("<tr>");
document.writeln("<td><br />Client Name and Address:<br /><br /></td>");
document.writeln("</tr>");

document.writeln("<tr>");
document.writeln("<td>" + memberName + "<br />" + streetAddress + "<br />" + city + ", " + province + " " + postalCode + "</td>");
document.writeln("</tr>");

document.writeln("<tr>");
document.writeln("<td>Phone: " + phoneNumber + " (H)<br />" + "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;" + cellNumber + " (C)</td>");
document.writeln("</tr>");

document.writeln("<tr>");
document.writeln("<td>Site #: " + siteNumber + 
    "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"
    + "Member type: ");

if (membershipType === "S") {
    document.writeln("Standard");
} else {
    document.writeln("Executive");
}
document.writeln("</td>");
document.writeln("</tr>");


document.writeln("<tr>");
document.writeln("<td>Alternate members: " + "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;" + alternateMembers + "<br />" +
    "Weekly site cleaning:" + "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;" + (weeklyCleaning === "Y" ? "Yes" : "No") + "<br />" +
    "Video surveillance:" + "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;" +(videoSurveillance === "Y" ? "Yes" : "No") + "</td>");
document.writeln("</tr>");

document.writeln("<tr>");
document.writeln("<td>Site charges: " + "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;" + cur2Format.format(siteCharges) + "<br />" +
    "Extra charges:" + "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;" + cur2Format.format(extraCharges) + "<br />");
document.writeln("</tr>");

document.writeln("<tr>");
document.writeln("<td>Subtotal: " + "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;" + cur2Format.format(subTotal) + "<br />" +
    "Sales tax (HST):" + "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;" + cur2Format.format(tax) + "<br />");
document.writeln("</tr>");

document.writeln("<tr>");
document.writeln("<td>Total monthly charges: " + "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;" + cur2Format.format(totalMonthlyCharge) + "<br />" +
    "Monthly dues:" + "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;" + cur2Format.format(monthlyDues) + "<br />");
document.writeln("</tr>");

document.writeln("<tr>");
document.writeln("<td>Total monthly fees: " + "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;" + cur2Format.format(totalMonthlyFees) + "<br />" +
    "Total yearly fees:" + "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;" + cur2Format.format(totalYearlyFees) + "<br />");
document.writeln("</tr>");

document.writeln("<tr>");
document.writeln("<td>Monthly payment: " + "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;" + cur2Format.format(monthlyPayment) + "<br />");
document.writeln("</tr>");

document.writeln("<tr>");
document.writeln("<td>Issued: " + "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;" + currentDate + "<br />" + "<br />" +
    "HST Reg No:" + "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;" + "549-33-5849-47" + "<br />" + "<br />" +
    "Cancellation fee:" + "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;" + cur2Format.format(cancellationFee) + "<br />" + "</td>");
document.writeln("</tr>");

document.writeln("</table>");
document.writeln("<br />");

File: Q1/Web/codes/second half/yacht copy/yacht.py
# Description: A program to help St. John’s Marina & Yacht Club enter the appropriate
# information of their customers and prepare a receipt.
# Author: Abiodun Magret Oyedele & SD 14
# Date: Jan 26, 2025
 
# Define program constants.
NO_OF_SITES = 100
EVEN_NO_SITES_MONTHLY = 80
ODD_NO_SITES_MONTHLY = 120
ALTERNATE_MEMBER = 5
TAX_RATE = 0.15
MONTHLY_DUES_STANDARD = 75 
MONTHLY_DUES_EXECUTIVE = 150 
YEARLY_RATE = 12 
PROCESSING_FEE = 59.99 
CANCELLATION_RATE = 0.6
WEEKLY_SITE_CLEANING_RATE = 50
VIDEO_SURV_RATE = 35

# Gather user inputs.
entryDate = str(input("Enter date ((YYYY-MM-DD):  "))
siteNum = int(input("Enter site number (1-100): "))
memberName = str(input("Enter the customer name: "))
address = str(input("Enter the customer street address: "))
city = str(input("Enter the customer city: "))
province = str(input("Enter the customer povince: ").upper())
postalcode = str(input("Enter customer postal code: ").upper())
phoneNum = str(input("Enter the customer phone number: "))
cellNumber = str(input("Enter the customer cell number: "))
membershipType = str(input("Enter the customer membership type (S or E): ").upper())
alternateMem = int(input("Enter the number of customer alternate member: "))
weeklySiteClean = str(input("Weekly site cleaning option of customer (Y or N): ").upper())
videoSurv = str(input("Enter video surveillance option of customer (Y or N): ").upper())

# Perform required calculations.
if (siteNum % 2 == 1):
    siteCharges = ODD_NO_SITES_MONTHLY + (alternateMem * ALTERNATE_MEMBER)
else:
    siteCharges = EVEN_NO_SITES_MONTHLY + (alternateMem * ALTERNATE_MEMBER)

if (weeklySiteClean == "Y"):
    siteExtraCharge = WEEKLY_SITE_CLEANING_RATE
else: 
    siteExtraCharge = 0

if (videoSurv == "Y"): 
    videoExtraCharge = VIDEO_SURV_RATE
else: 
    videoExtraCharge = 0

extraCharges = siteExtraCharge  + videoExtraCharge
subTotal = siteCharges + extraCharges
tax = TAX_RATE * subTotal
totalMonthlyCharge = subTotal + tax

if (membershipType == "S"):
    totalMonthlyFees = totalMonthlyCharge + MONTHLY_DUES_STANDARD 
else: 
    totalMonthlyFees = totalMonthlyCharge + MONTHLY_DUES_EXECUTIVE

totalYearlyFees = totalMonthlyFees * YEARLY_RATE

monthlyPayment = (totalYearlyFees + PROCESSING_FEE) / YEARLY_RATE
cancellationFee = totalYearlyFees * CANCELLATION_RATE

# Display results
print()
print(f"       St. John’s Marina & Yacht Club")
print(f"            Yearly Member Receipt")
print()
print(f"____________________________________________")

print()
print(f" Client Name and Address:")
print(f" {memberName}")
print(f" {address}")
print(f" {city}, {province} {postalcode}")

print()
print(f" Phone:  {phoneNum} (H)")
print(f"         {cellNumber} (C)")

print()
if (membershipType == "S"):
    memberType = "Standard"
else:
    memberType = "Executive"

siteNumberDsp = "{:03d}".format(siteNum)
# print(f" Site #: {siteNumberDsp} Member type: {memberType:>9s}")
print(f" Site #: {siteNumberDsp}      Member type: {memberType:>13s}")

print()
# print(f" Alternate members: {alternateMem:>15d}")
print(f" Alternate members: {alternateMem:>24d}")

if (weeklySiteClean == "Y"):
    answer = "YES"
else:
    answer = "NO"
print(f" Weekly site cleaning: {answer:>21s}")

if (videoSurv == "Y"):
    answer1 = "YES"
else:
    answer1 = "NO"
print(f" Video surveillance: {answer1:>23s}")

print()
siteChargesDsp = "${:,.2f}".format(siteCharges)
print(f" Site charges: {siteChargesDsp:>29s}")

extraChargesDsp = "${:,.2f}".format(extraCharges)
print(f" Extra charges: {extraChargesDsp:>28s}")
print(f"                                    ________")

subTotalDsp = "${:,.2f}".format(subTotal)
print(f" Subtotal: {subTotalDsp:>33s}")

taxDsp = "${:,.2f}".format(tax)
print(f" Sales tax (HST): {taxDsp:>26s}")
print(f"                                    ________")

totalMonthlyChargeDsp = "${:,.2f}".format(totalMonthlyCharge)
print(f" Total monthly charges: {totalMonthlyChargeDsp:>20s}")

if (membershipType == "S"):
    monthlyDues = MONTHLY_DUES_STANDARD
else:
    monthlyDues = MONTHLY_DUES_EXECUTIVE
monthlyDuesDsp = "${:,.2f}".format(monthlyDues)
print(f" Monthly dues: {monthlyDuesDsp:>29s}")
print(f"                                    ________")

totalMonthlyFeesDsp = "${:,.2f}".format(totalMonthlyFees)
print(f" Total monthly fees: {totalMonthlyFeesDsp:>23s}")

totalYearlyFeesDsp = "${:,.2f}".format(totalYearlyFees)
print(f" Total yearly fees: {totalYearlyFeesDsp:>24s}")

print()
monthlyPaymentDsp = "${:,.2f}".format(monthlyPayment)
print(f" Monthly payment: {monthlyPaymentDsp:>26s}")

print()
print(f"                                    ________")

print()
print(f" Issued: {entryDate}")
print(f" HST Reg No: 549-33-5849-4720-9885")

print()
cancellationFeeDsp = "${:,.2f}".format(cancellationFee)
print(f" Cancellation fee: {cancellationFeeDsp:>25s}")

# Any housekeeping duties at the end of the program.
